src/transformers/trainer_pt_utils.py

```diff
@@ -1395,3 +1395,211 @@ def get_lr(self):
+                forward pass.
+        limit_all_gathers (`bool`, *optional*, defaults to `False`)
+            FSDP's limit_all_gathers (useful only when `fsdp` field is passed).
+                If `"True"`, FSDP explicitly synchronizes the CPU thread to prevent too many in-flight
+                all-gathers.
+        use_orig_params (`bool`, *optional*, defaults to `True`)
+            If `"True"`, allows non-uniform `requires_grad` during init, which means support for interspersed
+            frozen and trainable paramteres. Useful in cases such as parameter-efficient fine-tuning. Please
+            refer this
+            [blog](https://dev-discuss.pytorch.org/t/rethinking-pytorch-fully-sharded-data-parallel-fsdp-from-first-principles/1019
+        sync_module_states (`bool`, *optional*, defaults to `True`)
+            If `"True"`, each individually wrapped FSDP unit will broadcast module parameters from rank 0 to
+            ensure they are the same across all ranks after initialization
+        cpu_ram_efficient_loading (`bool`, *optional*, defaults to `False`)
+            If `"True"`, only the first process loads the pretrained model checkpoint while all other processes
+            have empty weights.  When this setting as `"True"`, `sync_module_states` also must to be `"True"`,
+            otherwise all the processes except the main process would have random weights leading to unexpected
+            behaviour during training.
+        activation_checkpointing (`bool`, *optional*, defaults to `False`):
+            If `"True"`, activation checkpointing is a technique to reduce memory usage by clearing activations of
+            certain layers and recomputing them during a backward pass. Effectively, this trades extra
+            computation time for reduced memory usage.
+        state_dict_type (`str`, *optional*, defaults to `FULL_STATE_DICT`):
+            State dict type to use. If a string, it must be one of `full_state_dict`, `local_state_dict`, or `sharded_state_dict`
+        xla (`bool`, *optional*, defaults to `False`):
+            Whether to use PyTorch/XLA Fully Sharded Data Parallel Training. This is an experimental feature
+            and its API may evolve in the future.
+        xla_fsdp_settings (`dict`, *optional*)
+            The value is a dictionary which stores the XLA FSDP wrapping parameters.
+
+            For a complete list of options, please see [here](
+            https://github.com/pytorch/xla/blob/master/torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py).
+        xla_fsdp_grad_ckpt (`bool`, *optional*, defaults to `False`):
+            Will use gradient checkpointing over each nested XLA FSDP wrapped layer. This setting can only be
+            used when the xla flag is set to true, and an auto wrapping policy is specified through
+            fsdp_min_num_params or fsdp_transformer_layer_cls_to_wrap.
+        xla_fsdp_v2 (`bool`, *optional*, defaults to `False`):
```

Comment:
### Code change comments


Exposed this option since its being used in the trainer - https://github.com/huggingface/transformers/blob/03c12d0d63f781eda525900978f69eebc3df7478/src/transformers/trainer.py#L717 

and was being set though not exposed previously - https://github.com/huggingface/transformers/blob/03c12d0d63f781eda525900978f69eebc3df7478/src/transformers/training_args.py#L1900